flow/connectors/postgres/qrep.go

```diff
@@ -295,6 +312,85 @@ func (c *PostgresConnector) getNumRowsPartitions(
 	}
 }
 
+func (c *PostgresConnector) getCTIDBlockPartitions(
+	ctx context.Context,
+	tx pgx.Tx,
+	parsedWatermarkTable utils.SchemaTable,
+	numPartitions int64,
+	last *protos.QRepPartition,
+) ([]*protos.QRepPartition, error) {
+	if numPartitions == 0 {
+		return nil, fmt.Errorf("numPartitions must be greater than 0")
+	}
+
+	blocksQuery := "SELECT (pg_relation_size(to_regclass($1)) / current_setting('block_size')::int)::bigint"
```

Comment:
Any reason to not go for min(ctid), max(ctid)? Like with the example from the test the bottom half will be dead and min..max will actually cover the full range

Reply:
`ctid` is not indexed so calculating min/max would require a sequential scan, so this approach bypasses this for performance reasons. While it's less precise than `select min(ctid), max(ctid) from <table>` (i.e. it includes deletes), it's much faster.